File: 0462-minimum-moves-to-equal-array-elements-ii/0462-minimum-moves-to-equal-array-elements-ii.cpp
class Solution {
public:
    int minMoves2(vector<int>& nums) {
       sort(begin(nums),end(nums));
        int ans =0;
        vector<long long>v(nums.size());
        for(int i=0;i<nums.size();++i){
            ans+=nums[i];
            v[i]=ans;
        }
        long long res = INT_MAX;
        for(int i=0;i<nums.size();++i){
            long long x = ((long long)nums[i]*(i+1))-v[i];
            x+=(ans-v[i]) - (nums[i]*(nums.size()-i-1));
            res = min(res,x);
        }
        return res;
    }
};

File: 0462-minimum-moves-to-equal-array-elements-ii/0462-minimum-moves-to-equal-array-elements-ii.py
class Solution:
    def minMoves2(self, nums: List[int]) -> int:
        nums.sort()
        ans =0
        v = []
        for i in range (len(nums)):
            ans += nums[i]
            v.append(ans)
        res = 1e18
        for i in range (len(nums)):
            x = (nums[i]*(i+1))- v[i]
            x += ((ans - v[i]) - (nums[i] * (len(nums)-i-1)))
            res = min(res,x)
        return res
            